backend/src/controllers/items-controller.ts

```diff
@@ -96,13 +98,22 @@ export const getSpecificItem = async (req: Request, res: Response) => {
 }
 
 //Updates a specific item
-export const updateItem = async (req: Request, res: Response) => {
+export const updateItem = async (req: CustomRequest, res: Response) => {
     const { newName }: { newName: string }= req.body;
     const { newDescription }: { newDescription: string }= req.body;
     const { newPickupLocation }: { newPickupLocation: string }= req.body;
     const { newPickupTime }: { newPickupTime: string }= req.body;
     const itemID: string = req.params._id;
-
+    const ownerId = req.body.itemOwnerId;
```

Comment:
Similar to the `createItem`,
- Remove the `ownerId` code
- Use `req.token.userId` as the item owner ID
- Only let the user update the item at `itemId` if its itemOwnerId is `req.token.userId`